core/src/components/datetime/utils/manipulation.ts

```diff
@@ -361,5 +366,55 @@ export const validateParts = (parts: DatetimeParts): DatetimeParts => {
     partsCopy.day = numDays;
   }
 
+  /**
+   * If value is same day as min day,
+   * make sure the time value is in bounds.
+   */
+  if (minParts !== undefined && isSameDay(partsCopy, minParts)) {
+    /**
+     * If the hour is out of bounds,
+     * set it to the min hour.
+     * We use <= so that we can still check
+     * the minute even if the partsCopy hour
+     * is already the same as the minParts hour.
+     */
+    if (partsCopy.hour !== undefined && minParts.hour !== undefined && partsCopy.hour <= minParts.hour) {
+      partsCopy.hour = minParts.hour;
+
+      /**
+       * If the minute is out of bounds,
+       * set it to the min minute.
+       */
+      if (partsCopy.minute !== undefined && minParts.minute !== undefined && partsCopy.minute < minParts.minute) {
+        partsCopy.minute = minParts.minute;
+      }
```

Comment:
Playing around with this made me notice something. Say you have a datetime with a `min` of 5/1 at 8AM. You select a value of 5/3 at 4:30AM. You then click on 5/1. With the current logic, because the hour and minute are handled separately, the time is auto-adjusted to 8:30AM.

This feels strange to me; from a user perspective, I would expect the time to go to the min (8AM), since that's the closest possible time to my selection. Maybe the times should be compared as a whole instead? (This would also apply to the `max` logic below.)

Reply:
Good point 👍  I changed this in https://github.com/ionic-team/ionic-framework/pull/26125/commits/0f7db03f203dcafd0bf9539c4741522ccea0ca76.